cinema/urls.py

```diff
@@ -1,10 +1,24 @@
-from django.urls import path
+from django.urls import path, include
+from rest_framework import routers
 
-from cinema.views import movie_list, movie_detail
+from cinema.views import (
+    ActorList,
+    ActorDetail,
+    MovieViewSet,
+    CinemaHallViewSet,
+    GenreList,
+    GenreDetail)
+
+router = routers.DefaultRouter()
+router.register("movies", MovieViewSet)
+router.register("cinema_halls", CinemaHallViewSet)
 
 urlpatterns = [
-    path("movies/", movie_list, name="movie-list"),
-    path("movies/<int:pk>/", movie_detail, name="movie-detail"),
+    path("genres/", GenreList.as_view(), name="genre-list"),
+    path("genres/<int:pk>/", GenreDetail.as_view(), name="genre-detail"),
+    path("actors/", ActorList.as_view(), name="actor-list"),
+    path("actors/<int:pk>/", ActorDetail.as_view(), name="actor-detail"),
+    path("", include(router.urls)),
```

Comment:
No need to include if basepath is not exist. You can use: urlpatterns = [list of urls] + router.urls